ep/build-arrow/src/get_arrow.sh

```diff
@@ -29,16 +29,22 @@ for arg in "$@"; do
 done
 
 function checkout_code {
+  TARGET_BUILD_COMMIT="$(git ls-remote $ARROW_REPO $ARROW_BRANCH | awk '{print $1;}')"
   ARROW_SOURCE_DIR="$CURRENT_DIR/../build/arrow_ep"
   if [ -d $ARROW_SOURCE_DIR ]; then
     echo "Arrow source folder $ARROW_SOURCE_DIR already exists..."
     cd $ARROW_SOURCE_DIR
     git init .
-    git fetch $ARROW_REPO tag $ARROW_BRANCH
+    EXISTS=$(git show-ref refs/tags/build_$TARGET_BUILD_COMMIT || true)
+    if [ -z "$EXISTS" ]; then
+      git fetch $ARROW_REPO $TARGET_BUILD_COMMIT:refs/tags/build_$TARGET_BUILD_COMMIT
+    fi
     git reset --hard HEAD
+    git checkout refs/tags/build_$TARGET_BUILD_COMMIT
   else
     git clone $ARROW_REPO -b $ARROW_BRANCH $ARROW_SOURCE_DIR
     cd $ARROW_SOURCE_DIR
+    git checkout $TARGET_BUILD_COMMIT
```

Comment:
is it `git checkout refs/tags/build_$TARGET_BUILD_COMMIT`?

Reply:
`refs/tags/build_$TARGET_BUILD_COMMIT` will only be created in the previous if branch, running into the second branch means it's a brand-new build, we can ensure the current branch contains the target commit, so `so git checkout $TARGET_BUILD_COMMIT` can be enough.